### PR TITLE
feat(core): add area chart

### LINKED ISSUE #188: Area Chart
#### I am submitting a...

- [X] Feature request
- [ ] Design defect
- [ ] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

### LINKED ISSUE #253: Restyle: area chart
New style of Line chart is below. Reference text in `#D6316C` for measurements. Overall, all areas are stroked at the top with a `1px` line of legend color. All areas are filled with legend color at `.3 opacity`. Stacked areas are separated by a `1px` white divider line.

- [ ] Standard area 
- [ ] Stacked area

(3) Area charts

### LINKED ISSUE #590: Area chart not supported
#### I am submitting a...

- [x] Feature request

### PR DESCRIPTION
**Pull request by theiliad**
Closes #253 (we'll need a separate issue for stacked area)
Closes #590
Closes #188

Thanks to @werdnanoslen who got started on this 🙌 

**Update: waiting on #499**

### CODE REVIEW

**Comment by theiliad** [comment]
> When I update data there's some weirdness with some of the areas:
> 
> image

That's working fine. It's more of a design decision of how we want to make that look better, since in the screenshot you posted the 2 areas do not have a range between Thursday 03 & Friday 04

File: packages/core/src/components/graphs/area.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+// D3 Imports
+import { area } from "d3-shape";
+import { select } from "d3-selection";
+
+export class Area extends Component {
+	type = "area";
+
+	init() {
+		// Highlight correct scatter on legend item hovers
```
Comment (natashadecoste) [inline packages/core/src/components/graphs/area.ts]:
```suggestion
		// Highlight associated area on legend item hovers
```

File: packages/core/src/configuration.ts [hunk 2]
```diff
@@ -166,6 +167,15 @@ const lineChart: LineChartOptions = Tools.merge({}, axisChart, {
 	}
 } as LineChartOptions);
 
+/**
+ * options specific to line charts
```
Comment (cal-smith) [inline packages/core/src/configuration.ts]:
```suggestion
 * options specific to area charts
```

**Comment by natashadecoste** [comment]
I think we need to do something about the overflow of axis labels on the right side _at some point._ The chart isn't looking centered. Its unfortunate that an area chart looks really odd when using the `addSpaceOnEdges` since that would do the trick

image

**Comment by theiliad** [comment]
> Unless the nodes rest directly on the grid lines, you are unable to activate the tooltip to see the node values on hover.

We're going to raise that as an issue to work through. The issue spreads beyond area chart to almost any chart that supports continuous scales

**Comment by theiliad** [comment]
If no one has an objection here I'm going to merge area charts (once I add wrappers), and track the tooltip/axis bugs in a separate issue

**Comment by lissali** [comment]
will we support to add area in line or bar chart to show baseline as :
image

image

**Comment by cal-smith** [comment]
> will we support to add area in line or bar chart to show baseline as :
> image
> 
> image

This should be captured in an issue 👍 this PR is specifically for adding the Area chart type

File: packages/angular/src/charts.module.ts [hunk 3]
```diff
@@ -2,6 +2,13 @@ import { NgModule } from "@angular/core";
 import { CommonModule } from "@angular/common";
 
 import { BaseChart } from "./base-chart.component";
+<<<<<<< HEAD
+
+import { AreaChartComponent } from "./area-chart.component";
+import { DonutChartComponent } from "./donut-chart.component";
+import { PieChartComponent } from "./pie-chart.component";
+=======
+>>>>>>> 57f4f1d3fb0cb4d0fe905ca98319b263d4a13379
```
Comment (cal-smith) [inline packages/angular/src/charts.module.ts:11]: Looks like you committed some merge conflicts 😬

**Comment by natashadecoste** [comment]
looks like theres some issues on legend filtering
image
if you filter out all the datasets then filter them back on and then filter a second time, you will see the fill doesnt update with the outlines

happens with both the demo area charts
image

**Comment by theiliad** [comment]
> LGTM!
> 
> The tooltip interactions are super weird though ... It feels very fiddly to find the points ... I feel like a persistent bar as I'm mousing over + a snap as I get close to a point with a tooltip would be more usable - I can open an issue for that if it's something that's open for discussion

yes I'm actually looking to make the snap to grid functionality more persistent

**Comment by cal-smith** [comment]
> yes I'm actually looking to make the snap to grid functionality more persistent

Awesome!

File: packages/core/src/components/graphs/area.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import { area } from "d3-shape";
+
+export class Area extends Component {
+	type = "area";
+
+	init() {
+
+		const eventsFragment = this.services.events;
+
+		// Highlight correct circle on legend item hovers
```
Comment (natashadecoste) [inline packages/core/src/components/graphs/area.ts:16]:
```suggestion
		// Highlight correct area on legend item hovers
```

File: packages/core/src/components/graphs/area.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+		eventsFragment.addEventListener(
+			Events.Legend.ITEM_HOVER,
+			this.handleLegendOnHover
+		);
+
+		// Un-highlight circles on legend item mouseouts
```
Comment (natashadecoste) [inline packages/core/src/components/graphs/area.ts:22]:
```suggestion
		// Un-highlight area on legend item mouseouts
```

File: packages/core/src/components/graphs/area.ts [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+import { CartesianOrientations, Events } from "../../interfaces";
+
+// D3 Imports
+import { area } from "d3-shape";
+
+export class Area extends Component {
+	type = "area";
+
+	init() {
+
```
Comment (cal-smith) [inline packages/core/src/components/graphs/area.ts:13]: extra space here
```suggestion
```

Comment (theiliad) [reply packages/core/src/components/graphs/area.ts:13]: wow just noticed this. I'll add with the stacked area PR

**Pull request merged**